extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -61,8 +63,31 @@ class AmpIframe extends AMP.BaseElement {
         minTop);
   }
 
+  /**
+   * Transforms the srcdoc attribute if present to an equivalent data URI.
+   *
+   * It may be OK to change this later to leave the `srcdoc` in place and
+   * instead ensure that `allow-same-origin` is not present, but this
+   * implementation has the right security behavior which is that the document
+   * may under no circumstances be able to run JS on the parent.
+   */
+  transformSrcDoc() {
+    var srcdoc = this.element.getAttribute('srcdoc');
+    if (!srcdoc) {
+      return;
+    }
+    var sandbox = this.element.getAttribute('sandbox');
+    assert(
+        !((' ' + sandbox + ' ').match(/\s+allow-same-origin\s+/)),
+        'allow-same-origin is not allowed with the srcdoc attribute %s.',
+        this.element);
+    var dataUri = 'data:text/html;charset=utf-8;base64,' + btoa(srcdoc);
+    this.element.setAttribute('src', dataUri);
```

Comment:
If all the same, I'd rather if `amp-iframe` kept the original srcdoc and transformation would only be done to set on `iframe` element.